controllers/clusterversion_controller.go

```diff
@@ -436,3 +450,42 @@ func (c *ClusterVersionReconciler) ensureConsolePlugin() error {
 
 	return nil
 }
+
+func (c *ClusterVersionReconciler) getDeployCSIConfig() (bool, error) {
+
+	// verify existence of storagecluster crd
+	storageClusterCRD := &metav1.PartialObjectMetadata{}
+	storageClusterCRD.SetGroupVersionKind(
+		extv1.SchemeGroupVersion.WithKind("CustomResourceDefinition"),
+	)
+	storageClusterCRD.Name = "storageclusters.ocs.openshift.io"
+	if err := c.get(storageClusterCRD); kerrors.IsNotFound(err) {
+		// storagecluster crd doesn't exist and we need to proceed with deploying CSI
+		return true, nil
+	} else if err != nil {
+		return false, fmt.Errorf("failed to verify existence of storagecluster crd: %v", err)
+	}
+
+	// storagecluster crd exists and we need to wait for explicitly deployment set in the configmap
+	operatorConfig := &corev1.ConfigMap{}
+	operatorConfig.Name = operatorConfigMapName
+	operatorConfig.Namespace = c.OperatorNamespace
+	if err := c.get(operatorConfig); err != nil {
+		return false, fmt.Errorf("failed to get operator configmap: %v", err)
+	}
```

Comment:
Don't we have this one loaded already on the reconciler?

Reply:
no.